### PR TITLE
Upgrade the Gradle wrapper to 7.6.1, with enabling the Java toolchain

### PR DESCRIPTION
**Pull request by hiroyuki-sato**
* Upgrade the Gradle wrapper to 7.6.1, with enabling the Java toolchain
* Use a modern dependency lock file.
* Update embulk-{core|junit4|deps|spi} to 0.10.49
* Update embulk-util-config to 0.3.3
* Update embulk-util-json to 0.2.1
* Update embulk-util-timestamp to 0.2.2
* Update gradle-embulk-plugins to 0.5.5
* Update embulk-formatter-csv: to 0.10.42
* Update embulk-input-file: to 0.10.42
* Update embulk-output-file: to 0.10.42
* Update embulk-parser-csv: to 0.10.42

Replacing lockfile using the following commands.

```
for i in embulk-output-* ; do ./gradlew ${i}:dependencies --write-locks ; done
fd embulkPluginRuntime.lockfile | xargs rm -f
```

### CODE REVIEW

**Comment by hiroyuki-sato** [comment]
@dmikurube, Could you take a look when you get a chance?

File: build.gradle [hunk 1]
```diff
@@ -59,19 +58,20 @@ subprojects {
...
-        compile "org.embulk:embulk-util-json:0.1.1"
-        compile "org.embulk:embulk-util-timestamp:0.2.1"
-        compile "org.embulk:embulk-util-retryhelper:0.8.2"
-
-        testCompile "org.embulk:embulk-junit4:0.10.31"
-        testCompile "org.embulk:embulk-core:0.10.31"
-        testCompile "org.embulk:embulk-deps:0.10.31"
+        implementation "com.fasterxml.jackson.core:jackson-annotations:2.6.7"
+        implementation "com.fasterxml.jackson.core:jackson-core:2.6.7"
+        implementation "com.fasterxml.jackson.core:jackson-databind:2.6.7"
```
Comment (dmikurube) [inline build.gradle:63]: We confirmed that we can use `jackson-databind:2.6.7.5` safely here as it has totally the same set of classes with `jackson-datadbind:2.6.7`.

File: embulk-output-mysql/src/test/java/org/embulk/output/mysql/AfterLoadTest.java [hunk 2]
```diff
@@ -32,7 +32,8 @@
 
 public class AfterLoadTest
 {
-    private static final String BASIC_RESOURCE_PATH = "org/embulk/output/mysql/test/expect/after_load/";
+    private static final String BASIC_RESOURCE_NAME = "org/embulk/output/mysql/test/expect/after_load/";
+    private static final String BASIC_RESOURCE_PATH = "/" + BASIC_RESOURCE_NAME;
```
Comment (dmikurube) [inline embulk-output-mysql/src/test/java/org/embulk/output/mysql/AfterLoadTest.java:36]: It seems hard to tell the difference between `NAME` and `PATH`. How about renaming them to `..._RELATIVE_PATH` and `..._ABSOLUTE_PATH`?

**Review by dmikurube — APPROVED** [review]
It basically looks good, then approved, but let me leave a couple of comments.

In addition, let's wait for comments from @hito4t. What do you think?

File: build.gradle (hunk 1 above)
Comment (hiroyuki-sato) [reply build.gradle]: Thanks fixed in [85b6772](https://github.com/embulk/embulk-output-jdbc/pull/317/commits/85b6772cc22920d44fb582315db208a36f468c5c)

File: embulk-output-mysql/src/test/java/org/embulk/output/mysql/AfterLoadTest.java (hunk 2 above)
Comment (hiroyuki-sato) [reply embulk-output-mysql/src/test/java/org/embulk/output/mysql/AfterLoadTest.java]: Thanks fixed in [ae2ec04](https://github.com/embulk/embulk-output-jdbc/pull/317/commits/ae2ec04a4f68ef20c35181f0642ca722846e4b8d)

Comment (hito4t) [reply embulk-output-mysql/src/test/java/org/embulk/output/mysql/AfterLoadTest.java]: `TestingEmbulk.loadYamlResource` is calling `EmbulkTests.readResource`, and `EmbulkTests.readResource` is calling `Resources.getResource` .

Is it not possible to pass the same argument (`BASIC_RESOURCE_ABSOLUTE_PATH` or `BASIC_RESOURCE_RELATIVE_PATH`) to all three methods?

**Comment by hiroyuki-sato** [comment]
@dmikurube @hito4t I removed the Guava and unified the variable name. Could you recheck this PR when you get a chance?

File: embulk-output-postgresql/src/test/java/org/embulk/output/postgresql/PostgreSQLTests.java [hunk 3]
```diff
@@ -38,10 +37,15 @@ public static void execute(String sql)
         int code;
         try {
             Process process = pb.start();
-            ByteStreams.copy(process.getInputStream(), System.out);
+            InputStream inputStream = process.getInputStream();
+            byte[] buffer = new byte[8192];
+            int readSize;
+            while ((readSize = inputStream.read(buffer)) != -1) {
+                System.out.write(buffer,0,readSize);
```
Comment (dmikurube) [inline embulk-output-postgresql/src/test/java/org/embulk/output/postgresql/PostgreSQLTests.java:44]: The coding style convention prefers to have a whitespace (`' '`) after comma (`','`).

File: embulk-output-sqlserver/src/test/java/org/embulk/output/sqlserver/SQLServerTests.java [hunk 4]
```diff
@@ -46,31 +46,37 @@ public static void execute(String sql, String... options)
...
+                config.get(String.class, "host"),
+                "-d",
+                config.get(String.class, "database"),
+                "-Q",
+                sql
+                ));
+
         for (String option : options) {
-            args.add(option);
+            args.add((String)option);
```
Comment (dmikurube) [inline embulk-output-sqlserver/src/test/java/org/embulk/output/sqlserver/SQLServerTests.java:63]: The coding style convention prefers to have a whitespace (`' '`) after the casting parenthesis (`(String)`).

File: embulk-output-sqlserver/src/test/java/org/embulk/output/sqlserver/SQLServerTests.java [hunk 5]
```diff
@@ -46,31 +46,37 @@ public static void execute(String sql, String... options)
...
         pb.redirectErrorStream(true);
         int code;
         try {
             Process process = pb.start();
-            ByteStreams.copy(process.getInputStream(), System.out);
+            InputStream inputStream = process.getInputStream();
+            byte[] buffer = new byte[8192];
+            int readSize;
+            while ((readSize = inputStream.read(buffer)) != -1) {
+                System.out.write(buffer,0,readSize);
```
Comment (dmikurube) [inline embulk-output-sqlserver/src/test/java/org/embulk/output/sqlserver/SQLServerTests.java:75]: The coding style convention prefers to have a whitespace (`' '`) after comma (`','`).

**Review by dmikurube — APPROVED** [review]
Still lgtm! Thanks for working on it. Left a couple of nitpicking coding-style comments, but approved.

**Comment by hiroyuki-sato** [comment]
Thank you for the comments! Fixed coding styles!

**Review by hito4t — APPROVED** [review]
LGTM!

**Comment by hiroyuki-sato** [comment]
@dmikurube , @hito4t Thanks! Could you merge this commit? (I don't have permission). Do you think I should squash this PR?

**Comment by hiroyuki-sato** [comment]
Thanks!

**Pull request merged**